validator/client/propose.go

```diff
@@ -330,3 +355,81 @@ func (v *validator) getGraffiti(ctx context.Context, pubKey [48]byte) ([]byte, e
 
 	return []byte{}, nil
 }
+
+// processPandoraShardHeader method does the following tasks:
+// - Get pandora block header, header hash, extraData from remote pandora node
+// - Validate block header hash and extraData fields
+// - Signs header hash using a validator key
+// - Submit signature and header to pandora node
+func (v *validator) processPandoraShardHeader(ctx context.Context, beaconBlk *ethpb.BeaconBlock,
+	slot types.Slot, epoch types.Epoch, pubKey [48]byte) (bool, error) {
+
+	fmtKey := fmt.Sprintf("%#x", pubKey[:])
+	// Request for pandora chain header
+	header, headerHash, extraData, err := v.pandoraService.GetShardBlockHeader(ctx)
+	if err != nil {
+		log.WithField("blockSlot", slot).WithError(err).Error("Failed to request block from pandora node")
+		if v.emitAccountMetrics {
+			ValidatorProposeFailVec.WithLabelValues(fmtKey).Inc()
+		}
+		return false, err
+	}
+	// Validate pandora chain header hash, extraData fields
+	if err := v.verifyPandoraHeader(beaconBlk, slot, epoch, header, headerHash, extraData); err != nil {
+		log.WithField("blockSlot", slot).WithError(err).Error("Failed to validate pandora block header")
+		if v.emitAccountMetrics {
+			ValidatorProposeFailVec.WithLabelValues(fmtKey).Inc()
+		}
+		return false, err
+	}
+	headerHashSig, err := v.keyManager.SignHeaderHash(ctx, &validatorpb.SignRequest{
+		PublicKey:       pubKey[:],
+		SigningRoot:     headerHash[:],
+		SignatureDomain: nil,
+		Object:          nil,
+	})
+	if err != nil {
+		log.WithField("blockSlot", slot).WithError(err).Error("Failed to sign pandora header hash")
+		if v.emitAccountMetrics {
+			ValidatorProposeFailVec.WithLabelValues(fmtKey).Inc()
+		}
+		return false, err
+	}
+	header.MixDigest = common.BytesToHash(headerHashSig)
+	var headerHashSig32Bytes [32]byte
+	copy(headerHashSig32Bytes[:], headerHashSig)
+	return v.pandoraService.SubmitShardBlockHeader(ctx, header.Nonce.Uint64(), headerHash, headerHashSig32Bytes)
+}
+
+// verifyPandoraHeader verifies header hash and extraData field
+func (v *validator) verifyPandoraHeader(beaconBlk *ethpb.BeaconBlock, slot types.Slot, epoch types.Epoch,
```

Comment:
either we hard `...ShardHeader` everywhere, or nowhere. :) see function above
Lets better call it `verifyPandoraShardHeader`

Reply:
private functions sometimes are used as separation of certain piece of code, in my opinion as log as it is descriptive i'm fine.
Altough `shardHeader` could be nice convention to stay with